k6/generic/change_config_maps.js

```diff
@@ -1,70 +1,467 @@
+  churnOpsCounter.add(1, { resource: resource });
+  changeEvents += 1
+}
+
+export function list(cookies) {
+  if (api === "steve") {
+    const url = cluster === "local" ?
+      `${baseUrl}/v1/${resource}` :
+      `${baseUrl}/k8s/clusters/${cluster}/v1/${resource}`
+
+    if (paginationStyle === "k8s") {
+      listWithK8sStylePagination(url, cookies)
+    }
+    else if (paginationStyle === "steve") {
+      listWithSteveStylePagination(url, cookies)
+    }
+    else {
+      fail("Invalid PAGINATION_STYLE value: " + paginationStyle)
     }
+  }
+  else if (api === "norman") {
+    const url = `${baseUrl}/v3/${resource}`
+    listWithNormanStylePagination(url, cookies)
+  }
+  else {
+    fail("Invalid API value: " + api)
+  }
+}
+
+function listWithK8sStylePagination(url, cookies) {
+  let revision = null
+  let continueToken = null
+  while (true) {
+    const fullUrl = url + "?limit=" + pageSize +
+      (revision != null ? "&revision=" + revision : "") +
+      (continueToken != null ? "&continue=" + continueToken : "") +
+      urlSuffix
+
+    const res = http.get(fullUrl, { cookies: cookies })
+
+    const criteria = {}
+    criteria[`listing ${resource} from cluster ${cluster} (steve with k8s style pagination) succeeds`] = (r) => r.status === 200
+    criteria[`no slow pagination errors (410 Gone) detected`] = (r) => r.status !== 410
+    check(res, criteria)
+
+    try {
+      const body = JSON.parse(res.body)
+      if (body === undefined || body.continue === undefined || firstPageOnly) {
+        break
+      }
+      if (revision == null) {
+        revision = body.revision
+      }
+      continueToken = body.continue
+    } catch (e) {
+      if (e instanceof SyntaxError) {
+        fail("Response body does not parse as JSON: " + res.body)
+      }
+      throw e
+    }
+
+    pause()
+  }
+}
+
+function listWithSteveStylePagination(url, cookies) {
+  let i = 1
+  let revision = null
+  while (true) {
+    const fullUrl = url + "?pagesize=" + pageSize + "&page=" + i +
+      (revision != null ? "&revision=" + revision : "") +
+      urlSuffix
+
+    const res = http.get(fullUrl, { cookies: cookies })
+
+    const criteria = {}
+    criteria[`listing ${resource} from cluster ${cluster} (steve style pagination) succeeds`] = (r) => r.status === 200
+    criteria[`no slow pagination errors (410 Gone) detected`] = (r) => r.status !== 410
+    check(res, criteria)
+
+    try {
+      const body = JSON.parse(res.body)
+      if (body === undefined || body.data === undefined || body.length === 0 || firstPageOnly) {
+        break
+      }
+      if (revision == null) {
+        revision = body.revision
+      }
+      i = i + 1
+    }
+    catch (e) {
+      if (e instanceof SyntaxError) {
+        fail("Response body does not parse as JSON: " + res.body)
+      }
+      throw e
+    }
+
+    pause()
+  }
+}
+
+function listWithNormanStylePagination(url, cookies) {
```

Comment:
If we are tailoring this test to Vai, we should get rid of duplicated unused code like this.

If we are looking for an exact implementation of list as in api_benchmark.js, would it be an option to import and reuse from there instead?

Reply:
Yes, we should be able to import these instead (as long as we don't require unique changes to the `pause` mechanism for this test in the future).